node/src/accountant/scanners_utils.rs

```diff
@@ -74,33 +92,85 @@ pub mod payable_scanner_utils {
                     )
                 },
             );
-
         format!("Payable scan found {} debts; the biggest is {} owed for {}sec, the oldest is {} owed for {}sec",
                 all_non_pending_payables.len(), biggest.balance_wei, biggest.age,
                 oldest.balance_wei, oldest.age)
     }
 
-    pub fn separate_errors(
-        sent_payments: &SentPayables,
-        logger: &Logger,
-    ) -> (Vec<Payable>, Vec<BlockchainError>) {
-        sent_payments
-            .payable
+    pub fn separate_errors<'a, 'b>(
+        sent_payables: &'a SentPayables,
+        logger: &'b Logger,
+    ) -> (Vec<&'a PendingPayable>, Option<PayableTransactingErrorEnum>) {
+        match &sent_payables.payment_procedure_result {
+            Ok(individual_batch_responses) => {
+                if individual_batch_responses.is_empty() {
+                    panic!("Broken code: An empty vector of processed payments claiming to be an Ok value")
+                }
+                let (oks, err_hashes_opt) =
+                    separate_rpc_results(individual_batch_responses, logger);
+                let remote_errs_opt = err_hashes_opt.map(|e| RemotelyCausedErrors(e));
+                (oks, remote_errs_opt)
+            }
+            Err(e) => {
+                warning!(
+                    logger,
+                    "Any persisted data from failed process will be deleted. Caused by: {}",
+                    e
+                );
+
+                (vec![], Some(LocallyCausedError(e.clone())))
+            }
+        }
+    }
+
+    type SeparateTxsByResult<'a> = (Vec<&'a PendingPayable>, Vec<H256>);
```

Comment:
It might be clearer if you moved this definition down to just above `fold_guts()`.